Refactor

mlakatkou opened the pull request:
To ensure compatibility with new versions of dependencies and improve development experience.

mlakatkou committed:
[refaktor] Update dependencies

To ensure compatibility with new versions of dependencies and improve development experience.

mlakatkou committed:
Merge remote-tracking branch 'origin/main' into refactor

AlexKlimenkov commented on README.md:
````diff
@@ -11,14 +11,14 @@ Clone the repository or download files.
 ```
 yarn
-yarn start
+yarn dev
````
pls keep `yarn start` and `npm start` as run commands

AlexKlimenkov commented on package.json:
```diff
@@ -1,19 +1,19 @@
-    "start": "vite",
+    "dev": "vite",
     "build": "vite build",
-    "preview": "vite preview --port 4173"
```
keep `start` as start script and keep `--port` flag in `preview` command

AlexKlimenkov commented on src/components/EventCalendarComponent.vue:
```diff
@@ -0,0 +1,25 @@
+        });
+    },
+    unmounted() {
+        this.$refs.cont.innerHTML = "";
```
We should call the destructor from here, clearing the container won't release event handlers that may be attached to it or to window https://docs.dhtmlx.com/eventcalendar/api/methods/js_eventcalendar_destructor_method/
(we've missed it in the initial version of the demo)

AlexKlimenkov reviewed (changes requested):
Everything looks great, but need to revert changes to start commands and add a destructor call to unmount

mlakatkou committed:
[update] Change npm script

To improve development experience.

mlakatkou committed:
[update] Change run commands

To improve development experience.

mlakatkou committed:
[update] Add --port 4173 flag to package.json file

To allow listening on a specific port for production build previews.

mlakatkou committed:
[update] Add destructor method

To remove all HTML elements of the Event Calendar and detach all related events after the Event Calendar component has been unmounted from the DOM.
